### DISCUSSION TITLE
Don't let pub publishers delete their entire community.  Make community deletion more transparent

### DISCUSSION THREAD

**Discussion by metasj** [knowledgefutures/pubpub#1895] [Ideas]
Currently, anyone who can publish a pub can potentially delete their entire community (including, presumably, all pubs, community members, files, history, &c). This behavior may not match what community members expect of their editors (or site admins).  

- As an admin, this makes me worry about adding other admins.
- As a user who knows this is currently possible w/o notification, this makes me worry that my work in any given community will eventually disappear.

Proposal:
 1. Separate a technical admin role (which can trigger community deletion) from a publisher role (which can publish reviews + pubs).  In my experience the "delete everything" button can keep communities from seriously using PubPub as their platform of record, or from ever adding a second admin.
 2. Add a link to a page describing community deletion next to the "request community deletion" button.  Note how a deletion request is processed and how it is communicated to authors and members if the community has more than one member.   

<img width="987" alt="Screen Shot 2022-03-22 at 18 10 52" src="https://user-images.githubusercontent.com/1615548/159584784-62c621ae-0c08-4f16-858d-f929877783b3.png">

My expectation / hope:
* Community deletion is reversible (unless a community was large enough that its archiving was a burden)
* The option to begin the community deletion process can only be triggered by technical site administrators, and involves some notice to other users of the community.
* Published articles are archived somehow before deletion [w/ Internet Archive?] and community members have the option to migrate their personal pubs to another community.

**Comment by metasj** [comment]
I tried to write this to focus on a single bug: allowing anyone who can publish a pub to delete their whole community.  That may be a controversial take, as issues mentioning this keep being converted to ideas 😭 but it strikes me as true for a broad range of contexts. Few off-¶ editorial teams allow content-vetting editors to take down the site.  

Robustness to social failures is as important as robustness to technical ones.  

For the abstract idea-level discussion: there are ways (like #958) to allow the current 'admin' role be held by one or two technical admins, whose responsibilities might include keeping a community up and making backups in case it goes down.

**Comment by idreyn** [comment]
@metasj I completely agree that a `publish` role between `admin` and `manage` is desirable. I fear I misread the intent of your issue, as it includes other pretty open-ended questions about deletion reversibility and transparency, and so I converted it to a discussion. We try to reserve the "bug" label for errors and deviations from intended functionality, and to use the "enhancement" label for improvements and feature requests.

For now we've been content, if not altogether happy, with the "Request Community Deletion" flow for a few reasons:

- All deletion requests are manually reviewed by a member of the PubPub team, and we don't get them very frequently — almost always from users who are trying out PubPub and not finding it useful.
- I don't know if we've yet had a situation where it's ambiguous whether the person requesting the deletion has the authority to speak on behalf of their community. We'd likely seek consensus from users with `admin` privileges before proceeding. We also have to find a final resting place for any content that's been issued a DOI. Since this situation is [rare and hard to meaningfully automate](https://xkcd.com/1205/), we haven't built any tooling for it yet.
- We keep a few months' worth of backups of our content to guard against catastrophe, which includes mishaps/social engineering related to Community deletion. But after that, we're pretty serious about the idea of your data being gone from PubPub for good if you ask us to delete it. Rather than making deletion an always-reversible operation, we'll focus on lossless round-trip data portability.

It does help my perspective on this issue to know that people are explicitly asking about it, though!
